packages/reactive_ble_mobile/ios/Classes/ReactiveBle/Central.swift

```diff
@@ -82,17 +84,27 @@ final class Central {
                 )
             },
             onCharacteristicNotificationStateUpdate: papply(weak: self) { central, characteristic, error in
+                guard let q = try? CharacteristicInstance(characteristic)
+                else {
+                    return
+                }
+                
                 central.characteristicNotifyRegistry.updateTask(
-                    key: QualifiedCharacteristic(characteristic),
+                    key: q,
                     action: { $0.complete(error: error) }
                 )
             },
             onCharacteristicValueUpdate: papply(weak: self) { central, characteristic, error in
-                onCharacteristicValueUpdate(central, QualifiedCharacteristic(characteristic), characteristic.value, error)
+                onCharacteristicValueUpdate(central, try! CharacteristicInstance(characteristic), characteristic.value, error)
```

Comment:
Not clear to me why here you force try building the CharacteristicInstance and not on the other two methods.